wal/wal.go

```diff
@@ -206,12 +209,13 @@ func (w *WAL) ReadAll() (metadata []byte, state raftpb.HardState, ents []raftpb.
 	rec := &walpb.Record{}
 	decoder := w.decoder
 
+	var match bool
```

Comment:
please add todos to readall. i think we need to detect more errors and loosen the checking of match.